The update to socket2 v0.4 raised the MSRV to 1.46

CI didn't catch it because MSRV is tested only with default features.
```
$ cargo +1.45.2 c --features tcp
    Checking socket2 v0.4.0
    Checking tokio v1.4.0
error[E0658]: `match` is not allowed in a `const fn`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/socket2-0.4.0/src/lib.rs:156:9
    |
156 | /         match address {
157 | |             SocketAddr::V4(_) => Domain::IPV4,
158 | |             SocketAddr::V6(_) => Domain::IPV6,
159 | |         }
    | |_________^
    |
    = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information
```

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -51,7 +51,7 @@ jobs:
           - stable
           - beta
           - nightly
-          - 1.45.2
+          - 1.46
 
         os:
           - ubuntu-latest
@@ -66,6 +66,8 @@ jobs:
           - rust: nightly
             features: "--features full,nightly"
             benches: true
+          - rust: 1.46
+            features: "--features full"
 
     runs-on: ${{ matrix.os }}
 

